aspnetcore/security/authentication/facebook-logins.md

```diff
@@ -0,0 +1,122 @@
+
+## Creating the app in Facebook
+
+*  Navigate to [https://developers.facebook.com/apps](https://developers.facebook.com/apps) and log in. If you don't already have a Facebook account, use the *sign up* link on the login page to create one.
+
+* Tap **My Apps** in the upper right corner:
+
+![image](sociallogins/_static/FBMyApps.png)
+
+* Tap **+ Add a New App** and fill out the form to create a new app ID:
+
+![image](sociallogins/_static/FBNewAppID.png)
+
+* The **Product Setup** page will be displayed, letting you select the features your new app will support. Tap **Get Started** on *Facebook Login*:
+
+![image](sociallogins/_static/FBProductSetup.png)
+
+* You will be presented with the **Client OAuth Settings** page with some defaults already set:
+
+![image](sociallogins/_static/FBOAuthSetup.png)
+
+* Enter your base URI with *signin-facebook* appended into the **Valid OAuth Redirect URIs** field. For example, https://localhost:44320/**signin-facebook**.
+
+> [!NOTE]
+> You don't need to configure **signin-facebook** as a route in your app. The ASP.NET Core team's implementation of the OAuth flow will create a temporary socket (called a *backchannel*) that listens at this route just for the duration of the OAuth flow.
+
+* Make a note of your *App ID* and your *App Secret* before dismissing this page so that you can add both into your ASP.NET Core app later in this tutorial.
+
+* Tap **Save Changes** to complete the new application configuration.
+
+## Storing Facebook AppId and AppSecret
+
+Link sensitive settings like Facebook *AppID* and *Secret* to your application configuration by using the [Secret Manager tool](../app-secrets.md) instead of storing them in your configuration file directly, as described in the [previous section](sociallogins.md). Execute the following in your project working directory:
+
+* Set the Facebook AppId
+
+  <!-- literal_block {"ids": [], "xml:space": "preserve"} -->
+
+  ````
+  dotnet user-secrets set Authentication:Facebook:AppId <app-Id>
+     ````
+
+* Set the Facebook AppSecret
+
+  <!-- literal_block {"ids": [], "xml:space": "preserve"} -->
+
+  ````
+  dotnet user-secrets set Authentication:Facebook:AppSecret <app-secret>
+     ````
+
+The following code reads the configuration values stored by the [Secret Manager](../app-secrets.md#security-app-secrets):
+
+[!code-csharp[Main](../../common/samples/WebApplication1/Startup.cs?highlight=11&range=20-36)]
+
+## Enable Facebook middleware
+
+> [!NOTE]
+> You will need to use NuGet to install the [Microsoft.AspNetCore.Authentication.Facebook](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Facebook/1.1.0-preview1-final) package if it hasn't already been installed. Alternatively, execute the following in your project directory:
```

Comment:
This is kind of a problem. We want them to use the most recent version, but when linking to https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Facebook/ you get the page for the old version. Is there different URL action I don't know about that always points to latest, even if latest is marked as pre-release?